crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -121,12 +123,21 @@ impl Extension for QueryLimitsChecker {
     ) -> ServerResult<ExecutableDocument> {
         // TODO: limit number of variables, fragments, etc
         // TODO: limit/ban directives for now
-        // TODO: limit overall query size
 
         let cfg = ctx
             .data::<ServiceConfig>()
             .expect("No service config provided in schema data");
 
+        if query.len() > cfg.limits.max_query_payload_size as usize {
+            return Err(ServerError::new(
```

Comment:
This code needs to use `graphql_error` so that it returns an error code.

Reply:
The closest error code from the spec which makes sense is `GRAPHQL_VALIDATION_FAILED`, so added that.